File: OISGP/app/src/main/java/id/ois/gp/SettingActivity.java
package id.ois.gp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.Toast;

import id.ois.gp.database.MasterUrlDataSource;

public class SettingActivity extends AppCompatActivity {

    private MasterUrlDataSource dataSource;
    String url;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);

        dataSource = new MasterUrlDataSource(this);
        dataSource.open();

        String topUrl = dataSource.getTopUrl();
        if (topUrl != null) {
            url = topUrl;
        } else {
            url = "http://api-gp.farkhan.biz.id";
        }

        EditText editText = findViewById(R.id.editText);
        editText.setText(url);

        findViewById(R.id.buttonSave).setOnClickListener(
                v -> {

                    dataSource.updateUrl(1, editText.getText().toString());
                    Intent intent = new Intent(SettingActivity.this, MainActivity.class);
                    startActivity(intent);
                }
        );

    }
}

File: ValletApp/app/src/main/java/net/valet/view/SettingActivity.kt
package net.valet.view

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import net.valet.databinding.ActivitySettingBinding
import net.valet.helper.SessionManager
import net.valet.helper.SessionManagerApps
import net.valet.utils.Global

class SettingActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySettingBinding
    private lateinit var session: SessionManager
    private lateinit var sessionApps: SessionManagerApps

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingBinding.inflate(layoutInflater)
        setContentView(binding.root)

        session = SessionManager(this)
        sessionApps = SessionManagerApps(this)
        initComponent()
        initButton()
    }

    private fun initButton() {
        binding.btnSubmitTransact.setOnClickListener {
            Global.apiServer = binding.etApiServer.text.toString()
            sessionApps.apiServer = binding.etApiServer.text.toString()
            val intent = Intent(this@SettingActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    private fun initComponent() {
        binding.etApiServer.setText(sessionApps.apiServer)
    }



}